[BUG] service-latency panic with rds-core workload

Duplicated version of https://github.com/kube-burner/kube-burner-ocp/issues/312

# Bug Description

## **Output of `kube-burner-ocp` version**

```
# kube-burner-ocp version
Version: 1.7.5
Git Commit: 9e446137a086b3010c56026b3fc3647b403d1269
Build Date: 2025-09-05T15:38:10Z
Go Version: go1.23.12
OS/Arch: linux amd64
```

## **Describe the bug**

Am getting a panic when trying to run rds-core workload with service-latency

## **To Reproduce**

Steps to reproduce the behavior:

1. `kube-burner-ocp rds-core --service-latency=true --iterations=1 --qps=10 --burst=10 --dpdk-cores=2 --perf-profile workers-dpdk --churn-percent=10 --churn-duration=1h`

## **Expected behavior**

The job runs and collects data.

## **Screenshots or output**

```
# kube-burner-ocp rds-core --service-latency=true --iterations=1 --qps=10 --burst=10 --dpdk-cores=2 --perf-profile workers-dpdk --churn-percent=10 --churn-duration=1h
time="2025-09-11 23:17:56" level=info msg="❤️ Checking for Cluster Health" file="cluster_health.go:46"
time="2025-09-11 23:17:56" level=info msg="Cluster is Healthy" file="cluster_health.go:54"
time="2025-09-11 23:17:56" level=info msg="🔥 Starting kube-burner (1.7.5@9e446137a086b3010c56026b3fc3647b403d1269) with UUID ebc88c78-720d-46a2-ae6e-6bbb2698e0b7" file="job.go:91"
time="2025-09-11 23:17:56" level=info msg="📈 Registered measurement: podLatency" file="factory.go:109"
time="2025-09-11 23:17:56" level=info msg="📈 Registered measurement: serviceLatency" file="factory.go:109"
time="2025-09-11 23:17:56" level=info msg="QPS: 10" file="job.go:371"
time="2025-09-11 23:17:56" level=info msg="Burst: 10" file="job.go:378"
time="2025-09-11 23:17:56" level=info msg="Job bgp-setup: 1 iterations with 1 IPAddressPool replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="Job bgp-setup: 1 iterations with 1 BGPAdvertisement replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="Job bgp-setup: 1 iterations with 1 BGPPeer replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="QPS: 10" file="job.go:371"
time="2025-09-11 23:17:56" level=info msg="Burst: 10" file="job.go:378"
time="2025-09-11 23:17:56" level=info msg="Job rds: 1 iterations with 42 Secret replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="Job rds: 1 iterations with 30 ConfigMap replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="Job rds: 1 iterations with 1 NetworkPolicy replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="Job rds: 1 iterations with 1 NetworkPolicy replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="Job rds: 1 iterations with 1 NetworkPolicy replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="Job rds: 1 iterations with 2 SriovNetwork replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="Job rds: 1 iterations with 20 Service replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="Job rds: 1 iterations with 1 Service replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="Job rds: 1 iterations with 2 Route replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="Job rds: 1 iterations with 15 Deployment replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="Job rds: 1 iterations with 10 Deployment replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="Job rds: 1 iterations with 2 Deployment replicas" file="create.go:84"
time="2025-09-11 23:17:56" level=info msg="Pre-load: images from job bgp-setup" file="pre_load.go:73"
time="2025-09-11 23:17:56" level=info msg="No images found to pre-load, continuing" file="pre_load.go:79"
time="2025-09-11 23:17:56" level=info msg="Pre-load: images from job rds" file="pre_load.go:73"
time="2025-09-11 23:17:56" level=info msg="Pre-load: Creating DaemonSet using images [quay.io/cloud-bulldozer/nginx:latest quay.io/cloud-bulldozer/curl:latest ghcr.io/abraham2512/fedora-stress-ng:master] in namespace preload-kube-burner" file="pre_load.go:195"
time="2025-09-11 23:17:56" level=info msg="Pre-load: Sleeping for 15s" file="pre_load.go:86"
time="2025-09-11 23:18:11" level=info msg="Deleting 1 namespaces with label: kube-burner-preload=true" file="namespaces.go:67"
time="2025-09-11 23:18:36" level=info msg="Namespace kube-burner-service-latency already exists" file="namespaces.go:44"
time="2025-09-11 23:18:36" level=info msg="Creating /v1, Resource=pods latency watcher for bgp-setup" file="base_measurement.go:69"
time="2025-09-11 23:18:36" level=warning msg="pods \"svc-checker\" already exists" file="common.go:75"
time="2025-09-11 23:18:36" level=info msg="Creating /v1, Resource=services latency watcher for bgp-setup" file="base_measurement.go:69"
time="2025-09-11 23:18:37" level=info msg="Triggering job: bgp-setup" file="job.go:122"
time="2025-09-11 23:18:37" level=info msg="Deleting IPAddressPools labeled with kube-burner-job=bgp-setup in metallb-system" file="namespaces.go:55"
time="2025-09-11 23:18:37" level=info msg="Deleting BGPAdvertisements labeled with kube-burner-job=bgp-setup in metallb-system" file="namespaces.go:55"
time="2025-09-11 23:18:37" level=info msg="Deleting BGPPeers labeled with kube-burner-job=bgp-setup in metallb-system" file="namespaces.go:55"
time="2025-09-11 23:18:37" level=info msg="0/1 iterations completed" file="create.go:119"
time="2025-09-11 23:18:37" level=info msg="Waiting up to 4h0m0s for actions to be completed" file="create.go:169"
time="2025-09-11 23:18:37" level=info msg="Actions completed" file="waiters.go:76"
time="2025-09-11 23:18:37" level=info msg="Verifying created objects" file="utils.go:119"
time="2025-09-11 23:18:37" level=info msg="Job bgp-setup took 0s" file="job.go:191"
time="2025-09-11 23:18:37" level=info msg="Stopping measurement: podLatency" file="factory.go:150"
time="2025-09-11 23:18:37" level=info msg="Evaluating latency thresholds" file="metrics.go:48"
time="2025-09-11 23:18:37" level=info msg="Stopping measurement: serviceLatency" file="factory.go:150"
time="2025-09-11 23:18:37" level=info msg="Deleting 1 namespaces with label: kubernetes.io/metadata.name=kube-burner-service-latency" file="namespaces.go:67"
time="2025-09-11 23:18:44" level=info msg="Creating /v1, Resource=pods latency watcher for rds" file="base_measurement.go:69"
time="2025-09-11 23:18:49" level=info msg="Creating /v1, Resource=services latency watcher for rds" file="base_measurement.go:69"
time="2025-09-11 23:18:49" level=info msg="Triggering job: rds" file="job.go:122"
time="2025-09-11 23:18:49" level=info msg="Deleting 1 namespaces with label: kube-burner-job=rds" file="namespaces.go:67"
time="2025-09-11 23:18:56" level=info msg="Deleting SriovNetworks labeled with kube-burner-job=rds in openshift-sriov-network-operator" file="namespaces.go:55"
time="2025-09-11 23:18:56" level=info msg="Churning enabled" file="job.go:129"
time="2025-09-11 23:18:56" level=info msg="Churn cycles: 0" file="job.go:130"
time="2025-09-11 23:18:56" level=info msg="Churn duration: 1h0m0s" file="job.go:131"
time="2025-09-11 23:18:56" level=info msg="Churn percent: 10" file="job.go:132"
time="2025-09-11 23:18:56" level=info msg="Churn delay: 2m0s" file="job.go:133"
time="2025-09-11 23:18:56" level=info msg="Using deletion strategy: default" file="job.go:134"
time="2025-09-11 23:18:56" level=info msg="0/1 iterations completed" file="create.go:119"
E0911 23:19:06.723035 1372981 iface.go:275] "Observed a panic" panic="interface conversion: interface {} is *unstructured.Unstructured, not *v1.Service" panicGoValue="&runtime.TypeAssertionError{_interface:(*abi.Type)(0x26532c0), concrete:(*abi.Type)(0x2b0b600), asserted:(*abi.Type)(0x2aee820), missingMethod:\"\"}" stacktrace=<
        goroutine 896 [running]:
        k8s.io/apimachinery/pkg/util/runtime.logPanic({0x2f46928, 0x48a6d40}, {0x26ecc00, 0xc000db8a20})
                /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:107 +0xbc
        k8s.io/apimachinery/pkg/util/runtime.handleCrash({0x2f46928, 0x48a6d40}, {0x26ecc00, 0xc000db8a20}, {0x48a6d40, 0x0, 0x437465?})
                /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:82 +0x5e
        k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000685c00?})
                /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:59 +0x108
        panic({0x26ecc00?, 0xc000db8a20?})
                /opt/hostedtoolcache/go/1.23.12/x64/src/runtime/panic.go:791 +0x132
        k8s.io/client-go/listers.ResourceIndexer[...].Get(0x2f28500, {0xc000cc8016, 0x5})
                /home/runner/go/pkg/mod/k8s.io/client-go@v0.31.1/listers/generic_helpers.go:71 +0x12e
        github.com/kube-burner/kube-burner/pkg/measurements.(*serviceLatency).waitForIngress.func1({0x5f5e100?, 0xc000cc80a0?})
                /home/runner/go/pkg/mod/github.com/kube-burner/kube-burner@v1.17.4/pkg/measurements/service_latency.go:286 +0x54
        k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func1(0xc00108fbc0?, {0x2f46960?, 0x48a6d40?})
                /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/wait/loop.go:53 +0x52
        k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext({0x2f46960, 0x48a6d40}, {0x2f3a6d0, 0xc00108fbc0}, 0x1, 0x0, 0xc000cbb4a0)
                /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/wait/loop.go:54 +0x115
        k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel({0x2f46960, 0x48a6d40}, 0x7f9bf40f67544a76?, 0x1, 0xc000cbb4a0)
                /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/wait/poll.go:33 +0x56
        github.com/kube-burner/kube-burner/pkg/measurements.(*serviceLatency).waitForIngress(0x200000003?, 0x0?)
                /home/runner/go/pkg/mod/github.com/kube-burner/kube-burner@v1.17.4/pkg/measurements/service_latency.go:285 +0x46
        github.com/kube-burner/kube-burner/pkg/measurements.(*serviceLatency).handleCreateSvc.func1(0xc000c58788)
                /home/runner/go/pkg/mod/github.com/kube-burner/kube-burner@v1.17.4/pkg/measurements/service_latency.go:105 +0xa5
        created by github.com/kube-burner/kube-burner/pkg/measurements.(*serviceLatency).handleCreateSvc in goroutine 251
                /home/runner/go/pkg/mod/github.com/kube-burner/kube-burner@v1.17.4/pkg/measurements/service_latency.go:98 +0x27d
 >
panic: interface conversion: interface {} is *unstructured.Unstructured, not *v1.Service [recovered]
        panic: interface conversion: interface {} is *unstructured.Unstructured, not *v1.Service


goroutine 896 [running]:
k8s.io/apimachinery/pkg/util/runtime.handleCrash({0x2f46928, 0x48a6d40}, {0x26ecc00, 0xc000db8a20}, {0x48a6d40, 0x0, 0x437465?})
        /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:89 +0xee
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000685c00?})
        /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:59 +0x108
panic({0x26ecc00?, 0xc000db8a20?})
        /opt/hostedtoolcache/go/1.23.12/x64/src/runtime/panic.go:791 +0x132
k8s.io/client-go/listers.ResourceIndexer[...].Get(0x2f28500, {0xc000cc8016, 0x5})
        /home/runner/go/pkg/mod/k8s.io/client-go@v0.31.1/listers/generic_helpers.go:71 +0x12e
github.com/kube-burner/kube-burner/pkg/measurements.(*serviceLatency).waitForIngress.func1({0x5f5e100?, 0xc000cc80a0?})
        /home/runner/go/pkg/mod/github.com/kube-burner/kube-burner@v1.17.4/pkg/measurements/service_latency.go:286 +0x54
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func1(0xc00108fbc0?, {0x2f46960?, 0x48a6d40?})
        /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/wait/loop.go:53 +0x52
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext({0x2f46960, 0x48a6d40}, {0x2f3a6d0, 0xc00108fbc0}, 0x1, 0x0, 0xc000cbb4a0)
        /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/wait/loop.go:54 +0x115
k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel({0x2f46960, 0x48a6d40}, 0x7f9bf40f67544a76?, 0x1, 0xc000cbb4a0)
        /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/wait/poll.go:33 +0x56
github.com/kube-burner/kube-burner/pkg/measurements.(*serviceLatency).waitForIngress(0x200000003?, 0x0?)
        /home/runner/go/pkg/mod/github.com/kube-burner/kube-burner@v1.17.4/pkg/measurements/service_latency.go:285 +0x46
github.com/kube-burner/kube-burner/pkg/measurements.(*serviceLatency).handleCreateSvc.func1(0xc000c58788)
        /home/runner/go/pkg/mod/github.com/kube-burner/kube-burner@v1.17.4/pkg/measurements/service_latency.go:105 +0xa5
created by github.com/kube-burner/kube-burner/pkg/measurements.(*serviceLatency).handleCreateSvc in goroutine 251
        /home/runner/go/pkg/mod/github.com/kube-burner/kube-burner@v1.17.4/pkg/measurements/service_latency.go:98 +0x27d
```
## **Additional context**

errors in the traceback are from kube-burner@v1.17.4 so maybe the problem could be there too.

--- a/pkg/measurements/service_latency.go
+++ b/pkg/measurements/service_latency.go
@@ -51,7 +51,7 @@ var supportedServiceLatencyJobTypes = map[config.JobType]struct{}{
 
 type serviceLatency struct {
 	BaseMeasurement
-  stopInformerCh    chan struct{}
+	stopInformerCh    chan struct{}
 	epsLister         ldiscoveryv1.EndpointSliceLister
 	svcLister         lcorev1.ServiceLister
 	svcLatencyChecker *mutil.SvcLatencyChecker
@@ -206,7 +206,7 @@ func (s *serviceLatency) Start(measurementWg *sync.WaitGroup) error {
 	factory := informers.NewSharedInformerFactory(clientset, 0)
 	// EndpointSlice/Service lister & informer from typed client
 	s.epsLister = factory.Discovery().V1().EndpointSlices().Lister()
-    s.svcLister = factory.Core().V1().Services().Lister()
+	s.svcLister = factory.Core().V1().Services().Lister()
 	epsInformer := factory.Discovery().V1().EndpointSlices().Informer()
 	svcInformer := factory.Core().V1().Services().Informer()
 

